packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx

```diff
@@ -141,6 +163,16 @@ export const BaseSearchIndexModal: React.FunctionComponent<
     onSubmit(indexName, indexDefinitionDoc);
   }, [onSubmit, parsingError, indexName, indexDefinition]);
 
+  const onChangeTemplate = useCallback(
+    (template: SearchTemplate) => {
+      editorRef.current?.applySnippet(template.snippet);
+      rafraf(() => {
+        editorRef.current?.focus();
+      });
```

Comment:
Focusing after apply can mess with snippet logic, so you should rather apply after focus, both calls should be done from the next frame:

```suggestion
      rafraf(() => {
        editorRef.current?.focus();
        editorRef.current?.applySnippet(template.snippet);
      });
```

We learned it when integrating codemirror snippets with [select in aggregation builder](https://github.com/mongodb-js/compass/blob/70e22b25fdaf21b15fc45d4873cbb4804c0072c8/packages/compass-aggregations/src/components/stage.tsx#L172-L179)

Reply:
Will do the change, thanks a lot! I've found that without rafraf it did something weird sometimes, but didn't know that we should focus before applying the snippet.